doc/src/guides/solving/solve-equation-algebraically.md

```diff
@@ -0,0 +1,413 @@
+
+You can solve an equation in several ways. 
+The examples below demonstrate using both {func}`~.solve` and 
+{func}`~.solveset` where applicable. 
+You can choose the function best suited to your equation.
+
+### Make your equation into an expression that equals zero
+
+Use the fact that any expression not in an `Eq` (equation) is automatically 
+assumed to equal zero (0) by the solving functions. You can rearrange the 
+equation $x^2 = y$ to $x^2 - y = 0$, and solve that expression. This approach 
+is convenient if you are interactively solving an expression which already 
+equals zero, or an equation that you do not mind rearranging to 
+$expression = 0$.
+
+```py
+>>> from sympy import solve, solveset
+>>> from sympy.abc import x, y
+>>> solve(x ** 2 - y, x, dict=True)
+[{x: -sqrt(y)}, {x: sqrt(y)}]
+>>> solveset(x**2 - y, x)
+{-sqrt(y), sqrt(y)}
+```
+    
+### Put your equation into `Eq` form
+
+Put your equation into `Eq` form, then solve the `Eq`. 
+This approach is convenient if you are interactively solving an equation 
+which you already have in the form of an equation,
+or which you think of as an equality.
+    
+```py
+>>> from sympy import Eq, solve, solveset
+>>> from sympy.abc import x, y
+>>> eqn = Eq(x**2, y)
+>>> eqn
+Eq(x**2, y)
+>>> solutions = solve(eqn, x, dict=True)
+>>> print(solutions)
+[{x: -sqrt(y)}, {x: sqrt(y)}]
+>>> solutions_set = solveset(eqn, x)
+>>> print(solutions_set)
+{-sqrt(y), sqrt(y)}
+>>> for solution_set in solutions_set:
+...     print(solution_set)
+sqrt(y)
+-sqrt(y)
+```
+
+### Restrict the domain of solutions
+
+By default, SymPy will return solutions in the complex domain, which also 
+includes purely real and imaginary values. Here, the first two solutions are 
+real, and the last two are imaginary:
+
+```py
+>>> from sympy import Symbol, solve, solveset
+>>> x = Symbol('x')
+>>> solve(x ** 4 - 256, x, dict=True)
+[{x: -4}, {x: 4}, {x: -4*I}, {x: 4*I}]
+>>> solveset(x ** 4 - 256, x)
+{-4, 4, -4*I, 4*I}
+```
+
+To restrict returned solutions to real numbers, or another domain or range, 
+the different solving functions use different methods.
+
+For {func}`~.solve`, place an assumption on the symbol to be solved for, $x$
+
+```py
+>>> from sympy import Symbol, solve
+>>> x = Symbol('x', real=True)
+>>> solve(x ** 4 - 256, x, dict=True)
+[{x: -4}, {x: 4}]
+```
+
+or restrict the solutions with standard Python techniques for filtering a 
+list such as a list comprehension, or 
+by adding inequalities to {func}`~.solve` (but the range must be continuous):
+
+```py
+>>> from sympy import Or, Symbol, solve
+>>> x = Symbol('x', real=True)
+>>> expr = (x-4)*(x-3)*(x-2)*(x-1)
+>>> solution = solve(expr, x)
+>>> print(solution)
+[1, 2, 3, 4]
+>>> solution_outside_2_3 = [v for v in solution if (v.is_real and Or(v<2,v>3))]
+>>> print(solution_outside_2_3)
+[1, 4]
+>>> solution_2_3 = solve((expr,x>=2,x<=3), x)
+>>> print(solution_2_3)
+Eq(x, 2) | Eq(x, 3)
```

Comment:
We really shouldn't recommend this as a way of using `solve`. Just look at the weird output that it generates. This is almost certainly not what someone would have expected:
```python
In [1]: eq = (x-4)*(x-3)*(x-2)*(x-1)

In [2]: solve(eq, x)
Out[2]: [1, 2, 3, 4]

In [3]: solve((eq, x>=2, x<=3), x)
Out[3]: x = 2 ∨ x = 3
```
If we're going to show things like this then we need to explain why this happens. Really I would list this under a "caveats" section rather than as a tip for how to use solve.

Reply:
Removed this recommendation in #23805 due to confusing output formatting